### DISCUSSION TITLE
getRemoteAddressAsText always returns the same IP

### DISCUSSION THREAD

**Discussion by sametserpil** [uNetworking/uWebSockets#1804] [Q&A] [answered]
Hello,

I'm using uws behind nginx proxy. getRemoteAddressAsText() always return the same ip address for all clients connected to my server. I've checked the examples but couldn't find anything. Can you guys help me? My main goal is to restrict connections from the same IP to a certain number. How can I achieve that?

Thanks.

**Comment by slavslavov** [comment, marked as answer]
You need to build with UWS_WITH_PROXY. You will probably need to configure nginx to use the proxy protocol v2. Then use getProxiedRemoteAddressAsText() instead. I am using similar configuration with HAProxy and it works fine.
